openapi.yaml

```diff
@@ -47,33 +47,45 @@ info:
-    presented the scope levels the application is requesting. Once the User
-    accepts the request for access, we redirect them back to the application's
-    specified redirect URI with an access code.
+    Before you begin, you need to [Create an OAuth Client](https://developers.linode.com/api/v4/account-oauth-clients/#post).
 
-    Once the User has logged in to Linode and you have received an exchange code,
-    you will need to exchange that access code for an Authorization token. You
-    do this by making an HTTP POST request to the following address:
+      - For this you will pass a `label` and a `redirect_uri`.
+      - The return from this endpoint will give you a `client_id` and a `secret`.
+      - You can choose to make this `public`. The workflow below is split for public and private paths where they diverge.
 
-    ```
-    https://login.linode.com/oauth/token
-    ```
+    The OAuth workflow is a series of exchanges between your third-party app and Linode used
+    to authenticate a User before an application can start making API calls on the User's behalf.
+
+    | PRIVATE WORKFLOW | PUBLIC WORKFLOW |
+    |------------------|-----------------|
+    | 1.  The User visits the application's website and is directed to login with Linode. | 1.  The User visits the application's website and is directed to login with Linode. |
+    | 2.  Your application then redirects the user to Linode's [authentication server](https://login.linode.com) with the client application's `client_id` and requested OAuth `scope`, which appear in the URL of the login page. | 2.  Your application then redirects the user to Linode's [authentication server](https://login.linode.com) with the client application's `client_id` and requested OAuth `scope`, which appear in the URL of the login page. |
+    | 3.  The user logs into the authorization server with their username and password. | 3.  The user logs into the authorization server with their username and password. |
+    | 4.  The authorization server redirects the user to the specificed redirect URL with a temporary authorization `code` (exchange code) in the URL. | 4.  The authorization server redirects the user back to your applaction with an OAuth token in the URL hash. This is temporary and expires in 2 hours without a `refresh_token`. |
+    | 5.  The application issues a POST request (*see below*) to the authentication server with the exchange code, `client_id`, and the client application's `client_secret`. In return, the server returns an OAuth token and `refresh_token`. |  |
+    | 6.  The authentication server responds to the client application with a new OAuth `access_token`. `access_token` is set to expire in 2 hours if not refreshed. |  |
+    | 7.  The refresh token can be used to get a new OAuth `access_token` for another 2 hours by contacting the authentication server again with the `client_id`, `client_secret`, and `refresh_token`. | |
```

Comment:
You also get back a new refresh token, allowing you to continue to keep the session alive.

Reply:
updated to be more clear